### PR TITLE
home work lesson1

### PR DESCRIPTION
**Pull request by serhiikhrychov**
Home task from lesson 1 - done

### CODE REVIEW

File: lesson1/serhii_khrychov_hometask.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Foo
+  def initialize(uniq = :not_uniq)
+    @@instance_num ||= 0
+    @@instance_num += 1
+    @instance_num = @@instance_num
```
Comment (nonkor) [inline lesson1/serhii_khrychov_hometask.rb]: Don't you think that names `@instance_num` and `@@instance_num` are really confusing, do you?

File: lesson1/serhii_khrychov_hometask.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @uniq = uniq
+  end
+
+  def my_instance
+    return ":uniq" if @uniq == :uniq
```
Comment (nonkor) [inline lesson1/serhii_khrychov_hometask.rb]: Why do you wrap symbol `:uniq` into string?

File: lesson1/serhii_khrychov_hometask.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    @instance_num.even? ? ":even" : ":odd"
```
Comment (nonkor) [inline lesson1/serhii_khrychov_hometask.rb]: The same here. You can just return `:even` or `:odd` as symbols, not a strings.

File: lesson1/serhii_khrychov_hometask.rb (hunk 2 above)
Comment (nonkor) [inline lesson1/serhii_khrychov_hometask.rb]: Interesting decision. I thought you would use singleton method for `:uniq` instance. But your solution also works.

File: lesson1/serhii_khrychov_hometask.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+instances = []
+
+uniq_instance = Foo.new(:uniq)
+
+for i in 0..5
+  instances.push Foo.new
+end
+
+instances.each {|instance| puts instance.my_instance}
```
Comment (nonkor) [inline lesson1/serhii_khrychov_hometask.rb]: Add expected output as a comments.
Like,

``` ruby
# => :odd
# => :even
# and so on
```

**Review by nonkor — CHANGES_REQUESTED** [review]
Good work. Apply my suggestions, please.

File: lesson1/serhii_khrychov_hometask.rb (hunk 2 above)
Comment (serhiikhrychov) [reply lesson1/serhii_khrychov_hometask.rb]: Because we have this - "должен возвращать :even" "и :odd" in task. But I will remove it from my code

**Comment by serhiikhrychov** [comment]
@nonkor code updated

File: lesson1/serhii_khrychov_hometask.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    return :uniq if @uniq == :uniq
+    @instance_num.even? ? :even : :odd
+  end
+end
+
+instances = []
+
+uniq_instance = Foo.new(:uniq)
+
+for i in 0..5
```
Comment (nonkor) [inline lesson1/serhii_khrychov_hometask.rb]: Do not use `for` in ruby at all. I forgot to write this first time, but it's a bad practice. Your case could be resolved by [#times](https://ruby-doc.org/core-2.2.0/Integer.html#method-i-times)

File: lesson1/serhii_khrychov_hometask.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
+class Foo
+
```
Comment (nonkor) [inline lesson1/serhii_khrychov_hometask.rb]: delete empty line

**Review by nonkor — APPROVED** [review]
👍 
Apply small suggestions, squash commits into one and I could merge it.

**Comment by nonkor** [comment]
@serhiikhrychov why didn't you squash commits?

**Comment by serhiikhrychov** [comment]
@nonkor I used "squash and merge" button

**Comment by serhiikhrychov** [comment]
@nonkor I thought it should work

**Comment by nonkor** [comment]
ok, it works too, sorry, I didn't see it

**Comment by serhiikhrychov** [comment]
ok. Thanks for your feedback

**Pull request merged**